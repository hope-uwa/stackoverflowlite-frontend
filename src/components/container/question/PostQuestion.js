/* eslint-disable react/prop-types */
/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import Banner from '../banner/Banner';
import { postQuestion } from '../../../actions/allQuestions/allQuestionAction';
import store from '../../../store/configureStore';

export class PostQuestion extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      body: '',
    };
  }

  handleInputChanges = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { createQuestion, history, } = this.props;
    try {
      await createQuestion({ ...this.state });
      if (this.props.success) {
        return setTimeout(() => history.push('/', { prev: 'create' }), 1000);
      }
    } catch (error) {
      /* do nothing */
    }
  }

  render() {
    const { token } = store.getState().loginReducer;


    const display = token ? <div>
      <Banner title={'Question Pad'} display={'none'}/>
      <div className="question-box aligned-center row ml1 mr1 mt2 mb2">
        <form id="question-form" className="question-form mt1">
          <h3>Give it a Title:</h3>
          <input type="text" onChange={this.handleInputChanges}
            className="w100" name="title" id="title"/>
          <h3>Question:</h3>
          <textarea name="body" onChange={this.handleInputChanges}
            className="w100" id="body" cols="30" rows="10"
            style={{ overflowY: 'scroll' }}></textarea>
          <div className="align-right mt1 mb1">
            <button type="submit" onClick={this.handleSubmit} id="submitQuestion"
              className="btn btn-success btn-lg btn-shadow">Publish</button>
          </div>
        </form>
      </div>
    </div> : <Redirect to="/"/>;
    return display;
  }
}
PostQuestion.propTypes = {
  success: PropTypes.string,
  createQuestion: PropTypes.func
};

export const mapStateToProps = (state) => {
  return {
    success: state.createQuestionReducer.status
  };
};

export const mapDispatchToProps = {
  createQuestion: x => postQuestion(x)
};
export default connect(mapStateToProps, mapDispatchToProps)(PostQuestion);
