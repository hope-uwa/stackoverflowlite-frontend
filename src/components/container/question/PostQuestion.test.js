import React from 'react';
import { shallow } from 'enzyme';
import { PostQuestion, mapDispatchToProps, mapStateToProps } from './PostQuestion';

describe('landing page', () => {
  const getProps = () => (
    {
      success: '',
      createQuestion: () => {}
    }
  );
  it('snap shot', () => {
    const props = getProps(true);
    const component = shallow(<PostQuestion {...props}/>);
    expect(component).toMatchSnapshot();
  });

  it('should load viewQuestion', () => {
    const props = getProps();
    const e = {
      target: {
        value: ''
      }
    };
    const component = shallow(<PostQuestion {...props}/>);
    component.instance().handleSubmit(e);
    expect(component.instance().handleSubmit).toBeDefined();
  });
  it('should define handleInputChange', () => {
    const e = {
      target: {
        value: ''
      }
    };
    const props = getProps();
    const component = shallow(<PostQuestion {...props}/>);
    component.instance().handleInputChanges(e);
    expect(component.instance().handleInputChanges).toBeDefined();
  });

  //   Write this later
  //  it('should handle input changes successfully', () => {
  //     const wrapper = setup();
  //     wrapper.find('#answer').simulate(
  //       'change',
  //       {
  //         target: {
  //           name: 'body',
  //           value: 'this is an answer'
  //         }
  //       }
  //     );
  //     wrapper.find('#answerQuestion').simulate('click', { preventDefault() {} });
  //     expect(mockFn.mock.calls[0]).toEqual(undefined);
  //   });

  it('define mapStateToProps', () => {
    const state = { createQuestionReducer: { status: '' } };
    expect(mapStateToProps(state)).toBeDefined();
  });
  it('define mapDispatchToProps', () => {
    expect(mapDispatchToProps).toBeDefined();
    expect(mapDispatchToProps.createQuestion()).toBeDefined();
  });
});
