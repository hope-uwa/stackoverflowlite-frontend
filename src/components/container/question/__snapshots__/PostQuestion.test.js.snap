// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page snap shot 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;
